playground/config/plugins.js

```diff
@@ -1,7 +1,25 @@
-module.exports = () => ({
-  "meilisearch": {
-    // Uncomment to use, some tests will fail
-    // "api_key": "masterKey",
-    // "host": "http://localhost:7700"
+module.exports = () => {
+  // use process.env.NODE_ENV and if conditions
+  // to create a custom config for each environment
+  // console.
+  const env = process.env.NODE_ENV || 'development'
+
+  if (env === 'test') {
+    return {}
   }
-})
+  else if (env === 'production') {
+    return {
+      meilisearch: {
+        apiKey: "masterKey",
+        host: "http://localhost:7700"
+      }
+    }
+  }
+  return {
+    meilisearch: {
+      apiKey: "masterKey",
+      host: "http://localhost:7700"
+    }
+  }
+}
+
```

Comment:
All this logic is actually not required, I think it would make more sense to use the strapi environment handling (see [documentation](https://strapi.io/documentation/developer-docs/latest/setup-deployment-guides/configurations.html#optional-configurations)) like this:

`config/env/test/plugins.js`:
```js
module.exports = () => {
  return {
    meilisearch: {
      // Keys must be made explicit empty since they would otherwise be overwritten by the defaults
      apiKey: '',
      host: '',
    }
  }
}
```
`config/plugins.js`:
```js
module.exports = () => {
  return {
    meilisearch: {
      apiKey: 'masterKey',
      host: 'http://localhost:7700',
    }
  }
}
```

Reply:
I tried doing the following: 

/config/plugin.js
```js
module.exports = () => {
  return {}
}
```

`/config/test/config.js` => equivalent at no file
```js
module.exports = () => {
  return {}
}
```

no `/config/production/config.js` -> output same as `test/config.js`

`/config/development/config.js`
```js
module.exports = () => {
  return {
    meilisearch: {
      apiKey: "masterKey",
      host: "http://localhost:7700"
    }
  }
}
```

Test and production tests are now working properly

Reply:
So I made some more tests, you can remove all files and for example only have `plugin/config/test` or another.

Development is always the default value so if we only have `plugin/config/test` and no other file it will still work :)